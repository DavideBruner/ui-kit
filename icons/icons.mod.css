@import '../materials/colors.mod.css';

.theme-black {
  fill: var(--new-color-black);
}

.theme-grey {
  fill: var(--new-color-gray-60);
}

.theme-white {
  fill: var(--new-color-white);
}

.theme-blue {
  fill: var(--new-color-blue);
}

.theme-green {
  fill: var(--new-color-green-25);
}

.theme-apple-green {
  fill: var(--new-color-green);
}

.theme-orange {
  fill: var(--new-color-orange);
}

.theme-red {
  fill: var(--new-color-red);
}

svg.scale:not(:root) {
  width: 100%;
  height: auto;
}
