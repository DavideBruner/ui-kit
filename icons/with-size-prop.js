import { mapProps } from 'recompose';
import classnames from 'classnames';
import styles from './icons.mod.css';

export default mapProps(({ size, ...remainingProps }) => ({
  ...remainingProps,
  ...(size === 'scale'
    ? { className: classnames(remainingProps.className, styles.scale) }
    : {}),
  width: size === 'small' ? 16 : 24,
  height: size === 'small' ? 16 : 24,
}));
