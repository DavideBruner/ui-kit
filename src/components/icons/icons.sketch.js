import React from 'react';
import styled from '@emotion/styled';
import * as UIKit from 'ui-kit';
import { Suite, SketchSpec } from '../../../test/percy';

const Inline = styled.div`
  display: flex;
  flex-direction: row;
`;

const icons = Object.keys(UIKit).filter(thing => thing.endsWith('Icon'));

const sizes = ['small', 'medium', 'big', 'scale'];
const themes = [
  'black',
  'grey',
  'white',
  'blue',
  'green',
  'green-light',
  'orange',
  'red',
];

export default (
  <Suite>
    <SketchSpec label="Icons">
      {icons.map(iconName => {
        const Icon = UIKit[iconName];
        return sizes.map(size => (
          <div key={`${iconName}-${size}`}>
            <Inline>
              {themes.map(theme => (
                <div
                  key={theme}
                  data-sketch-symbol={`Icon/${iconName}/${size}/${theme}`}
                >
                  <Icon size={size} theme={theme} />
                </div>
              ))}
            </Inline>
          </div>
        ));
      })}
    </SketchSpec>
  </Suite>
);
