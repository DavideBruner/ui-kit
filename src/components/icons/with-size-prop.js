import { mapProps } from 'recompose';
import classnames from 'classnames';
import styles from './icons.mod.css';

const ICON_SIZES = {
  SMALL: 12,
  MEDIUM: 16,
  BIG: 24,
};

const getIconSize = size => {
  switch (size) {
    case 'small':
      return ICON_SIZES.SMALL;
    case 'medium':
      return ICON_SIZES.MEDIUM;
    case 'big':
      return ICON_SIZES.BIG;
    default:
      return ICON_SIZES.BIG;
  }
};

export default mapProps(({ size, ...remainingProps }) => ({
  ...remainingProps,
  ...(size === 'scale'
    ? { className: classnames(remainingProps.className, styles.scale) }
    : {}),
  width: getIconSize(size),
  height: getIconSize(size),
}));
