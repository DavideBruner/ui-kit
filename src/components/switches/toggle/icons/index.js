import ToggleButtonIcon from './toggle-button-default.svg';
import ToggleBarIcon from './toggle-bar-default.svg';

export default {
  ButtonDefault: ToggleButtonIcon,
  BarDefault: ToggleBarIcon,
};
