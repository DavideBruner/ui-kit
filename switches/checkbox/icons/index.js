// NOTE: explicit names for component names
import CheckboxDefaultIcon from './checkbox-default.svg';
import CheckboxDefaultHoverIcon from './checkbox-hover.svg';
import CheckboxDefaultErrorIcon from './checkbox-error.svg';
import CheckboxCheckedIcon from './checkbox-checked.svg';
import CheckboxCheckedHoverIcon from './checkbox-checked-hover.svg';
import CheckboxCheckedErrorIcon from './checkbox-checked-error.svg';
import CheckboxDisabledIcon from './checkbox-disabled.svg';
import CheckboxCheckedDisabledIcon from './checkbox-checked-disabled.svg';
import CheckboxIndeterminateIcon from './checkbox-indeterminate.svg';
import CheckboxIndeterminateHoverIcon from './checkbox-indeterminate-hover.svg';
import CheckboxIndeterminateDisabledIcon from './checkbox-indeterminate-disabled.svg';
import CheckboxIndeterminateErrorIcon from './checkbox-indeterminate-error.svg';

export default {
  Default: CheckboxDefaultIcon,
  Hover: CheckboxDefaultHoverIcon,
  Error: CheckboxDefaultErrorIcon,
  Checked: CheckboxCheckedIcon,
  CheckedHover: CheckboxCheckedHoverIcon,
  CheckedError: CheckboxCheckedErrorIcon,
  Disabled: CheckboxDisabledIcon,
  CheckedDisabled: CheckboxCheckedDisabledIcon,
  Indeterminate: CheckboxIndeterminateIcon,
  IndeterminateHover: CheckboxIndeterminateHoverIcon,
  IndeterminateDisabled: CheckboxIndeterminateDisabledIcon,
  IndeterminateError: CheckboxIndeterminateErrorIcon,
};
