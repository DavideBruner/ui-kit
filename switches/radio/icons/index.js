// NOTE: explicit names for component names
import RadioOptionDefaultIcon from './radio-option-default.svg';
import RadioOptionDefaultHoverIcon from './radio-option-hover.svg';
import RadioOptionCheckedIcon from './radio-option-checked.svg';
import RadioOptionCheckedHoverIcon from './radio-option-checked-hover.svg';
import RadioOptionDisabledIcon from './radio-option-disabled.svg';
import RadioOptionCheckedDisabledIcon from './radio-option-checked-disabled.svg';

export default {
  Default: RadioOptionDefaultIcon,
  Hover: RadioOptionDefaultHoverIcon,
  Checked: RadioOptionCheckedIcon,
  CheckedHover: RadioOptionCheckedHoverIcon,
  Disabled: RadioOptionDisabledIcon,
  CheckedDisabled: RadioOptionCheckedDisabledIcon,
};
