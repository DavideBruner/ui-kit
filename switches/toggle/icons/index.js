import withSizeProp from '../with-size-prop';
import ToggleBackgroundDefaultIcon from './toggle-normal-bkgr.svg';
import ToggleBackgroundHoverIcon from './toggle-active-hover-bkgr.svg';
import ToggleBackgroundDisabledIcon from './toggle-disable-bkgr.svg';
import ToggleBackgroundCheckedIcon from './toggled-bkgr.svg';
import ToggleBackgroundCheckedHoverIcon from './toggled-active-hover-bkgr.svg';
import ToggleBackgroundCheckedDisabledIcon from './toggled-disable-bkgr.svg';
import ToggleButtonDefaultIcon from './toggle-normal-bt.svg';
import ToggleButtonHoverIcon from './toggle-active-hover-bt.svg';
import ToggleButtonDisabledIcon from './toggle-disable-bt.svg';
import ToggleButtonCheckedIcon from './toggled-bt.svg';
import ToggleButtonCheckedHoverIcon from './toggled-active-hover-bt.svg';
import ToggleButtonCheckedDisabledIcon from './toggled-disable-bt.svg';

export default {
  BackgroundDefault: ToggleBackgroundDefaultIcon,
  BackgroundHover: ToggleBackgroundHoverIcon,
  BackgroundDisabled: ToggleBackgroundDisabledIcon,
  BackgroundChecked: ToggleBackgroundCheckedIcon,
  BackgroundCheckedHover: ToggleBackgroundCheckedHoverIcon,
  BackgroundCheckedDisabled: ToggleBackgroundCheckedDisabledIcon,
  ButtonDefault: withSizeProp(ToggleButtonDefaultIcon),
  ButtonHover: withSizeProp(ToggleButtonHoverIcon),
  ButtonDisabled: withSizeProp(ToggleButtonDisabledIcon),
  ButtonChecked: withSizeProp(ToggleButtonCheckedIcon),
  ButtonCheckedHover: withSizeProp(ToggleButtonCheckedHoverIcon),
  ButtonCheckedDisabled: withSizeProp(ToggleButtonCheckedDisabledIcon),
};
