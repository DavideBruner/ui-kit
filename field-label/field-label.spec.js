import React from 'react';
import { shallow } from 'enzyme';
import { WarningIcon } from '../icons';
import FlatButton from '../buttons/flat-button';
import IconButton from '../buttons/icon-button';
import FieldLabel from './field-label';

const createTestProps = customProps => ({
  title: 'Label Title',
  ...customProps,
});

describe('rendering', () => {
  let props;
  let wrapper;
  let titleWrapper;
  let hintWrapper;
  let titleIconWrapper;
  let hintIconWrapper;

  describe('with title', () => {
    beforeEach(() => {
      props = createTestProps();
      wrapper = shallow(<FieldLabel {...props} />);
      titleWrapper = wrapper.find('Label');
    });

    it('should pass label text', () => {
      expect(titleWrapper).toHaveProp('children', 'Label Title');
    });
  });

  describe('with hint', () => {
    describe('when hint is given', () => {
      beforeEach(() => {
        props = createTestProps({ hint: 'Label hint' });
        wrapper = shallow(<FieldLabel {...props} />);
        hintWrapper = wrapper.find({ className: 'hint' });
      });

      it('should be present', () => {
        expect(hintWrapper).toBeTruthy();
      });

      it('should contain the hint text', () => {
        expect(hintWrapper.render().text()).toEqual('Label hint');
      });
    });

    describe('when hint is not given', () => {
      beforeEach(() => {
        props = createTestProps();
        wrapper = shallow(<FieldLabel {...props} />);
      });

      it('should not be present', () => {
        expect(wrapper).not.toRender({ className: 'hint' });
      });
    });
  });

  describe('with `hasRequiredIndicator`', () => {
    describe('when connected input has required indicator', () => {
      beforeEach(() => {
        props = createTestProps({ hasRequiredIndicator: true });
        wrapper = shallow(<FieldLabel {...props} />);
        titleWrapper = wrapper.find('Label');
      });

      it('should pass hasRequiredIndicator', () => {
        expect(titleWrapper).toHaveProp('isRequiredIndicatorVisible', true);
      });
    });

    describe('when input has not required indicator', () => {
      beforeEach(() => {
        props = createTestProps({ hasRequiredIndicator: false });
        wrapper = shallow(<FieldLabel {...props} />);
        titleWrapper = wrapper.find('Label');
      });

      it('should not pass hasRequiredIndicator', () => {
        expect(titleWrapper).toHaveProp('isRequiredIndicatorVisible', false);
      });
    });
  });

  describe('with `button`', () => {
    describe('when button is given', () => {
      beforeEach(() => {
        props = createTestProps({ button: <IconButton size="big" /> });
        wrapper = shallow(<FieldLabel {...props} />);
        titleIconWrapper = wrapper.find(IconButton);
      });

      it('should display the icon', () => {
        expect(wrapper).toRender(IconButton);
      });

      it('should set the icon size', () => {
        expect(titleIconWrapper.prop('size')).toEqual('small');
      });
    });

    describe('when titleIcon is not given', () => {
      beforeEach(() => {
        props = createTestProps();
        wrapper = shallow(<FieldLabel {...props} />);
      });

      it('should not display an icon', () => {
        expect(wrapper.prop('titleIcon')).toBeUndefined();
      });
    });
  });

  describe('with `hintIcon`', () => {
    describe('when hintIcon is given', () => {
      beforeEach(() => {
        props = createTestProps({ hintIcon: <WarningIcon /> });
        wrapper = shallow(<FieldLabel {...props} />);
        hintIconWrapper = wrapper.find(WarningIcon);
      });

      it('should display the icon', () => {
        expect(wrapper).toRender(WarningIcon);
      });

      it('should set the icon size', () => {
        expect(hintIconWrapper.prop('size')).toEqual('medium');
      });
    });

    describe('when hintIcon is not given', () => {
      beforeEach(() => {
        props = createTestProps();
        wrapper = shallow(<FieldLabel {...props} />);
      });

      it('should not display an icon', () => {
        expect(wrapper.prop('hintIcon')).toBeUndefined();
      });
    });
  });

  describe('with `badge`', () => {
    describe('when badge is given', () => {
      beforeEach(() => {
        props = createTestProps({ badge: <FlatButton /> });
        wrapper = shallow(<FieldLabel {...props} />);
      });

      it('should display the badge', () => {
        expect(wrapper).toRender(FlatButton);
      });
    });
  });
});
