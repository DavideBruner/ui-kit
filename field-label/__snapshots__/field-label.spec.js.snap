// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering with hint when hint is given should match snapshot 1`] = `
<React.Fragment>
  <div
    className="label"
  >
    <Inline
      alignItems="flexStart"
      scale="xs"
    >
      <TextWrap>
        <Label
          isBold={true}
          isRequiredIndicatorVisible={false}
        >
          Label Title
        </Label>
      </TextWrap>
    </Inline>
  </div>
  <div
    className="hint"
  >
    <Inline
      alignItems="center"
      scale="xs"
    >
      <TextDetail>
        Label hint
      </TextDetail>
    </Inline>
  </div>
</React.Fragment>
`;
